runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -3188,14 +3214,29 @@ class INTERPRETER_CLASS
 					if (framesWalked > maxSize) {
 						framesWalked = maxSize;
 					}
+#if JAVA_SPEC_VERSION >= 19
+					if (framesWalked + continuationFramesWalked > maxSize) {
```

Comment:
Brackets required here, which leads me to believe that this code would be simpler if you just added the continuation frame count to framesWalked. The code above is limiting the output to maxSize, without taking the continuation frames into account.

Reply:
> this code would be simpler if you just added the continuation frame count to framesWalked

I don't think this would work because the vthread frames and continuation frames are from separate arrays.

> Brackets required here

Sorry I'm quite sure where the brackets should go. Here's why I did this: the first `if` limits the vthread frames to maxSize and then the second `if` limits the continuation frames.

Reply:
For the brackets, don't assume order of operations - here you have `a + b > c` which should be `(a + b) > c`. You have similar assumptions elsewhere where you do `a + b  * c`.

I will take a more thorough look at the overall change.

Reply:
I see, didn't realize that you were taking about this.